p2p/channel.go

```diff
@@ -244,6 +252,10 @@ func (c *channel) remoteReadLoop() {
 			return
 		}
 
+		c.once.Do(func() {
```

Comment:
You cannot use `once` twice. It's already used in `Close()`.

Reply:
You can and it works, since you are performing different action with it.

Reply:
https://play.golang.org/p/Gyyqn8EwQpI

Reply:
confirming it does not work.

Reply:
True, we figured out why it worked in my case. Actual channel close was not being executed after successful session.